maven-repository-manager/src/main/java/org/jboss/pnc/mavenrepositorymanager/RepositoryManagerDriver.java

```diff
@@ -270,14 +278,39 @@ private void setupBuildRepos(BuildExecution execution, Indy indy)
             // Global-level repos, for captured/shared artifacts and access to the outside world
             addGlobalConstituents(buildGroup, tempBuild);
 
-            // add extra repositories removed from poms by the adjust process
-            addExtraConstituents(execution.getArtifactRepositories(), id, buildContentId, indy, buildGroup);
+            // add extra repositories removed from poms by the adjust process and set in BC by user
+            List<ArtifactRepository> extraDependencyRepositories = extractExtraRepositoriesFromGenericParameters(genericParameters);
+            if(execution.getArtifactRepositories() != null) {
+                extraDependencyRepositories.addAll(execution.getArtifactRepositories());
+            }
+            addExtraConstituents(extraDependencyRepositories, id, buildContentId, indy, buildGroup);
 
             indy.stores().create(buildGroup, "Creating repository group for resolving artifacts in build: " + id
                     + " (repo: " + buildContentId + ")", Group.class);
         }
     }
 
+    List<ArtifactRepository> extractExtraRepositoriesFromGenericParameters(Map<String, String> genericParameters) {
+        String extraReposString = genericParameters.get(EXTRA_PUBLIC_REPOSITORIES_KEY);
+        if (extraReposString == null)
+            return new ArrayList<>();
+
+        return Arrays.stream(extraReposString.split("\n"))
+                .map((repoString) -> {
+                    try {
+                        String id = new URL(repoString)
+                                .getHost()
+                                .replaceAll("\\.", "-");
+                        return ArtifactRepository.build(id, id, repoString.trim(), true, false);
+                    } catch (MalformedURLException e) {
+                        logger.info("Malformed repository URL entered: " + repoString + ". Skipping.");
```

Comment:
For the master PR it would be useful to use the user log (warning) here, see
https://github.com/project-ncl/pnc/blob/master/indy-repository-manager/src/main/java/org/jboss/pnc/indyrepositorymanager/IndyRepositorySession.java#L91